com.sap.adt.abapcleaner.gui/src/com/sap/adt/abapcleaner/gui/FrmMain.java

```diff
@@ -185,43 +185,83 @@ public static void cleanAutomatically(CommandLineArgs commandLineArgs, PrintStre
 			}
 		}
 
-		// perform the cleanup
+		// check if single or multi file clean up
+		if (commandLineArgs.sourcePaths != null && commandLineArgs.sourcePaths.length > 0) {
+			cleanMultiAutomatically(commandLineArgs, out, profile);
+		} else {
+			cleanSingleSourceAutomatically(commandLineArgs, commandLineArgs.sourceCode, out, profile);
+		}
+	}
+	
+	private static void cleanMultiAutomatically(CommandLineArgs commandLineArgs, PrintStream out,
+			Profile profile) {
+		Persistency persistency = Persistency.get();
+		int baseSourcePathLength = commandLineArgs.sourceDir.length();
+	  	for (String sourcePath : commandLineArgs.sourcePaths) {
+			String sourceCode = persistency.readAllTextFromFile(sourcePath);
+
+			CleanupResult result = cleanAutomatically(sourceCode, commandLineArgs.abapRelease, commandLineArgs.cleanupRange, profile, commandLineArgs.showStats);
+		  	if (result == null) {
+				out.println("Cleanup for file " + sourcePath +" cancelled.");
+				continue;
+		  	} else if (result.hasErrorMessage()) {
+		  		out.println("Errors during clean-up of file: " + sourcePath);
+		  		out.println(result.errorMessage);
+		  		continue;
+		  	}
+
+		  	// calculate target file location

```

Comment:
`persistency.getFileName(sourcePath)` will only work if `--sourcedir` does not have a deep directory structure.
I tested the code always with something like that:

```
- /src
    |--- /search
           |-- file.clas.abap
```

So I would have the path `..\src\search\file.clas.abap`.
Calling just `persistency.getFileName(..)` would return only `file.clas.abap` and it would get created directly beneath the given target directory, when it should be created at `..\target\search\file.clas.abap`.

Reply:
Ah, now I get it :-) I somehow missed that you call persistency.getFilesInDirectory( … recursive = true). Hm, wouldn't you usually expect a "--recursive" option to allow the user to determine whether or not subfolders should be included then? 

Apart from that, you're right then, this could be kept. However, the ```if (…) … substring(1);``` should then better be moved to PersistencyBase.combinePath() before the ```result += path;``` line, so we'd have it in a central place. (Hope the tests like it)